relay-server/src/actors/spooler/mod.rs

```diff
@@ -636,17 +727,51 @@ impl BufferState {
                         "failed to spool the in-memory buffer to disk",
                     );
                 }
+                relay_log::trace!(
+                    "Transition to disk spool: # of envelopes = {}",
+                    disk.count.unwrap_or_default()
+                );
+
                 Self::Disk(disk)
             }
-            Self::Disk(disk) if disk.is_empty().await.unwrap_or_default() => {
-                Self::MemoryFileStandby {
-                    ram: InMemory::new(config.spool_envelopes_max_memory_size()),
-                    disk,
+            Self::Disk(mut disk) if Self::is_below_low_mem_watermark(config, &disk).await => {
+                match disk.delete_and_fetch_all(services).await {
```

Comment:
Is 30%, although low, an arbitrary number? What's the behavior if Relay attempts to unspool all envelopes to the memory when it goes into an outage and receives many envelopes?

What do you think about approaching this the other way around? Instead of "there now is x% of memory available, spool everything", a continuous check of "while there's a y% of memory left, unspool a few".

Reply:
I do not follow your comment.  If I get you correctly, then the relay should not unspool anything if in the outage mode 

Could you, please, put a code example what you mean, maybe to illustrate the problem?

Reply:
the message processing is sequential, if there will be a problem again, the Relay will start spool again and will switch to disk eventually